v2/logging.go

```diff
@@ -1,45 +1,23 @@
 package shuttle
 
 import (
-	"context"
-	"fmt"
-	"os"
-	"time"
+	"io"
+	"log/slog"
 )
 
-type Logger interface {
-	Info(s string)
-	Warn(s string)
-	Error(s string)
-}
-
-// SetLoggerFunc sets the function to be used to acquire a logger when go-shuttle logs.
-func SetLoggerFunc(fn func(ctx context.Context) Logger) {
-	getLogger = fn
-}
-
-var getLogger = func(_ context.Context) Logger { return &printLogger{} }
-
-type printLogger struct{}
-
-func (l *printLogger) Info(s string) {
-	fmt.Println(append(append([]any{}, "INFO - ", time.Now().UTC(), " - "), s)...)
-}
-
-func (l *printLogger) Warn(s string) {
-	fmt.Println(append(append([]any{}, "WARN - ", time.Now().UTC(), " - "), s)...)
-}
+// logging is disabled by default
+var log = slog.New(slog.NewTextHandler(io.Discard, nil))
 
-func (l *printLogger) Error(s string) {
-	fmt.Println(append(append([]any{}, "ERROR - ", time.Now().UTC(), " - "), s)...)
+func getLogger() *slog.Logger {
+	return log
 }
 
-func log(ctx context.Context, a ...any) {
-	if os.Getenv("GOSHUTTLE_LOG") == "ALL" {
-		l := getLogger(ctx)
-		if l == nil {
-			return
-		}
-		l.Info(fmt.Sprint(a...))
+// EnableHandler allows to set a custom slog.Handler to be used by the go-shuttle logger.
+// If handler is nil, the default slog logger will be used.
+func EnableHandler(handler slog.Handler) {
```

Comment:
This func name isn't working from a package perspective

`SetLogHandler` ?

Reply:
hm might have been a bad commit... updated the name to `SetLogHandler`